packages/xo-server/src/api/test.js

```diff
@@ -86,3 +89,32 @@ copyVm.resolve = {
   vm: ['vm', 'VM'],
   sr: ['sr', 'SR'],
 }
+
+// -------------------------------------------------------------------
+
+export async function xapiPoolChange({
+  currentPoolObject,
+  newPoolHostname,
+  newPoolObject,
+}) {
+  const xapi = this.getXapi(currentPoolObject)
+  const { _pool: currentPool } = xapi
+
+  xapi._setUrl({ ...xapi._url, hostname: newPoolHostname })
+  const [newPool] = await Promise.all([
+    // used to trigger the "update" event
+    xapi.getAllRecords('pool').then(array => array[0]),
+    fromEvent(xapi.objects, 'update'),
+  ])
+
+  assert(currentPool.$id !== newPool.$id)
+  assert.doesNotThrow(() => this.getXapi(newPoolObject))
+  assert.throws(() => this.getXapi(currentPoolObject))
+}
+
+xapiPoolChange.permission = 'admin'
+xapiPoolChange.params = {
+  currentPoolObject: { type: 'string' },
+  newPoolHostname: { type: 'string' },
+  newPoolObject: { type: 'string' },
```

Comment:
What is a *pool object*?

Reply:
A XO Object contained on the new Pool.

Reply:
Then why is it a string?

Reply:
It's the ID of the XO object. Should i call it XO resource instead ?